src/service/resources-impl.js

```diff
@@ -497,6 +498,18 @@ export class Resources {
     /** @private {?Array<!Resource>} */
     this.pendingBuildResources_ = [];
 
+    /** @private {?Array<!Resource>} */
+    this.pendingPreconnectResources_ = [];
+
+    // If we do early preconnects we delay them a bit. This is kind of
+    // an unfortunate trade off, but it seems faster, because the DOM
+    // operations themselves are not free and might delay
+    this.eventuallyCallPreconnects_ = debounce(
```

Comment:
`debounce` waits X ms after the latest call before doing anything, which means they could be waiting longer than 5ms to actually be called.